packages/flutter_hooks/lib/src/listenable.dart

```diff
@@ -128,3 +128,61 @@ class _UseValueNotifierHookState<T>
   @override
   String get debugLabel => 'useValueNotifier';
 }
+
+/// Subscribes to a [Listenable] and registers `listener` to the passed `listenable`.
+///
+/// See also:
+///   * [useListenable]
+///   * [useEffect]
+void useOnListenableChange(VoidCallback listener, Listenable listenable) {
+  use(_OnListenableChangeHook(listener, listenable));
+}
+
+class _OnListenableChangeHook extends Hook<void> {
+  _OnListenableChangeHook(this.listener, this.listenable)
+      : super(keys: [listenable]);
+
+  final VoidCallback listener;
+  final Listenable listenable;
+
+  @override
+  _OnListenableChangeHookState createState() => _OnListenableChangeHookState();
+}
+
+class _OnListenableChangeHookState
+    extends HookState<void, _OnListenableChangeHook> {
+  Dispose? disposer;
+
+  @override
+  void initHook() {
+    super.initHook();
+    scheduleEffect();
+  }
+
+  @override
+  void didUpdateHook(_OnListenableChangeHook oldHook) {
+    super.didUpdateHook(oldHook);
+
+    if (hook.keys == null) {
+      disposer?.call();
+      scheduleEffect();
+    }
+  }
+
+  @override
+  void build(BuildContext context) {}
+
+  @override
+  void dispose() => disposer?.call();
+
+  void scheduleEffect() {
+    hook.listenable.addListener(hook.listener);
+    disposer = () => hook.listenable.removeListener(hook.listener);
```

Comment:
Dispose logic will have to be updated based on the listener logic change mentioned above